packages/analytics-plugin-google-global-site-tag/src/browser.ts

```diff
@@ -0,0 +1,528 @@
+       */
+      let newCookieConfig: { [key: string]: unknown} = {};
+      if (config.cookieConfig !== undefined) {
+        Object.entries(config.cookieConfig).forEach(([key, value]) => {
+          newCookieConfig[camelToSnakeCase(key)] = value;
+        })
+      }
+
+      // Initialize tracker instance on page
+      if (!loadedInstances[instanceName]) {
+        let gtagConfig: { [key: string]: unknown } = {
+          cookie_domain: config.domain || 'auto',
+          send_page_view: config.sendPageView ? config.sendPageView : true,
+          allow_google_signals: config.allowGoogleSignals,
+          allow_ad_personalization_signals: config.allowAdPersonalizationSignals,
+          custom_map: customDimensions,
+          anonymize_ip: config.anonymizeIp,
+          ...newCookieConfig,
+        }
+
+        if (config.linker) {
+          gtagConfig.linker = config.linker;
+        }
+
+        /* set custom dimensions from user traits */
+        const user = instance.user() || {}
+        const traits = user.traits || {}
+
+        if (Object.keys(traits).length) {
+          window.gtag('set', traits);
+        }
+
+        window.gtag('config', trackingId, gtagConfig);
+
+        loadedInstances[instanceName] = true;
+      }
+    },
+
+    // Set parameter scope at user level with 'set' method
+    identify: (props: IProps) => {
+      const { payload, config } = props;
+      identifyVisitor(payload.userId, payload.traits, config)
+    },
+
+    // Set parameter scope at page level with 'config' method
+    page: ({ payload, config, instance }: IProps) => {
+      if (!window.gtag || !config.trackingId) return;
+
+      const { properties } = payload;
+      const { resetCustomDimensionsOnPage, customDimensions } = config;
+      const campaign = instance.getState('context.campaign');
+
+      /* If dimensions are specified to reset, clear them before page view */
+      if (resetCustomDimensionsOnPage && resetCustomDimensionsOnPage.length && customDimensions) {
+        const resetDimensions = resetCustomDimensionsOnPage.reduce((acc: { [key: string]: unknown }, key: string) => {
+          if (customDimensions[key]) {
+            acc[key] = null // { 'dimension_name': null } etc
+          }
+          return acc
+        }, {})
+
+        if (Object.keys(resetDimensions).length) {
+          // Reset custom dimensions
+          window.gtag('set', resetDimensions);
+        }
+      }
+
+      /**
+       * Create pageview-related properties
+       */
+      const path = properties.path || document.location.pathname;
+      const pageView = {
+        page_path: path,
+        page_title: properties.title,
+        page_location: properties.url,
+      }
+
+      const campaignData = addCampaignData(campaign);
+      /**
+       * Creates property dimensions for global scope
+       */
+      const globalProperties = getProperties(properties, config.globalProperties);
+
+      const pageData = {
```

Comment:
To keep the consistency with `analytics-plugin-google-analytics` I include `path` in both `pageView` and `pageData`. I'm a bit confused why page_path is set in both `pageData` and `pageView`? The former is sent with `pageView event` and the latter is sent via `set`